crates/cxx-qt-build/src/lib.rs

```diff
@@ -506,6 +525,25 @@ impl CxxQtBuilder {
             cc_builder_whole_archive.file(qtbuild.qrc(&qrc_file));
             cc_builder_whole_archive_files_added = true;
         }
+
+        // If we are using Qt 5 then write the std_types source
+        // This registers std numbers as a type for use in QML
+        //
+        // Note that we need this to be compiled into the whole_archive builder
+        // as they are stored in statics in the source.
+        if qtbuild.version().major == 5 {
+            let std_types_contents = include_str!("std_types_qt5.cpp");
+            let std_types_path = format!(
+                "{out_dir}/std_types_qt5.cpp",
+                out_dir = env::var("OUT_DIR").unwrap()
+            );
+            let mut source =
+                File::create(&std_types_path).expect("Could not create std_types source");
```

Comment:
Do we need to do this elaborate copying of the C++ file? Can we just compile the C++ file directly without copying it first?

Reply:
Not sure we can, as the `CARGO_MANIFEST_DIR` turns out to be the example rather than the cxx-qt-build crate ?

Once we have support for +bundle and +whole-archive then we should be able to put this in `cxx-qt` then we should be able to remove the copying. Unless there is another way...

Reply:
Even using `file!()` doesn't give you an absolute path so it'd be quite tricky to do i think ?

Reply:
Left a comment saying that this can be improved once we have +bundle and +whole-archive.